pkg/profiler/profiler_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package profiler
+
+import (
+	"syscall"
+	"testing"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+
+	"github.com/stretchr/testify/require"
+)
+
+// The intent of these tests is to ensure that the BPF library we use,
+// (libbpfgo in this case) behaves in the way we expect.
+
+func SetUpBpfProgram() (*bpf.Module, error) {
+	m, err := bpf.NewModuleFromBufferArgs(bpf.NewModuleArgs{
+		BPFObjBuff: bpfObj,
+		BPFObjName: "parca",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = m.BPFLoadObject()
+	if err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+func TestDeleteNonExistentKeyReturnsEnoent(t *testing.T) {
+	m, err := SetUpBpfProgram()
+	require.NoError(t, err)
+	defer m.Close()
```

Comment:
```suggestion
	t.Cleanup(m.Close) 
```
nit: better for the tests. (I'm not exactly sure about above would work as is, please check locally.)

Reply:
TIL, thanks!!